templates/common/openstack/units/afterburn-hostname.service.yaml

```diff
@@ -11,6 +11,8 @@ contents: |
   Before=node-valid-hostname.service
 
   [Service]
+  # Wait for the metadata service to be accessible
+  ExecStartPre=/bin/sh -c 'i=1; while [ $i -le 300 ]; do ping -c1 -W1 169.254.169.254 > /dev/null && break; i=$(( $i + 1 )); done'
```

Comment:
Would `Restart=on-failure` be simpler and more robust here? This would require `openstack-afterburn-hostname` to correctly return a failure code on failure, which it probably does but I haven't checked.

Reply:
Almost certainly. I didn't think of that. Lemme try.